### PR TITLE
 Restore Previous Test Options Behavior (for Release Branch)

### LINKED ISSUE #19058: Consider reverting breaking change about test configuration
### Expected Behavior

As discussed with @ljacomet, the change to test framework configuration documented [here](https://github.com/gradle/gradle/issues/18606) is a major breaking change. It should probably have been a warning instead.

### Current Behavior

Upgrading a passing build to Gradle 7.3 breaks. See https://github.com/micronaut-projects/micronaut-gradle-plugin/issues/296
Any build generated by Micronaut in the past which doesn't configure a test framework explicitly is going to fail when upgrading to Gradle 7.3. Basically it means most of the builds generated, via launch.micronaut.io or the CLI tool, since if you say nothing Micronaut will use JUnit Platform..

### Context

The problem is that builds which used to pass, even if we understand that they were probably not configured as expected, now break. We have multiple users all reporting the same issue. In addition, it's actually pretty hard to figure out _why_ this is happening. In other words, it would be nice if Gradle captured where the `options` block is configured, so that instead of failing:

- it produces a _warning_ that the previous configuration will be discarded
- it tells _what_ configured the old `options`, so that we can trace back to what code was triggered too early

Later, possibly in 8.0, make this an error.

### Steps to Reproduce 

See https://github.com/micronaut-projects/micronaut-gradle-plugin/issues/296 for a reproducer and workaround.

### PR DESCRIPTION
**Pull request by tresat**
Fixes #19058

### Context
This is a continuation of [this PR](https://github.com/gradle/gradle/pull/19080), with the work rebased against the release branch.  Please see that PR for many explanatory comments.

### CODE REVIEW

**Comment by tresat** [comment]
Ignore the 74 in the branch naming, it should be 731.

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesIntegrationTest.groovy [hunk 1]
```diff
@@ -572,4 +576,78 @@ class TestSuitesIntegrationTest extends AbstractIntegrationSpec {
...
     }
+
+    def "multiple getTestingFramework() calls on a test suite return same instance"() {
+        given:
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            def first = testing.suites.test.getTestingFramework()
```
Comment (ljacomet) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesIntegrationTest.groovy:587]: Your method rename did not propagate here, I believe that's the test failures you are seeing.

**Review by ljacomet — APPROVED** [review]
LGTM!

One thing is to fix the test failures, see below:

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesIntegrationTest.groovy (hunk 1 above)
Comment (tresat) [reply subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesIntegrationTest.groovy]: That definitely broke this test, thanks.

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**